common.py

```diff
@@ -42,26 +71,28 @@ def checkout_validation_webpage_branch(self):
         - self.GIT_PULL_REQUEST: The state of the GIT CI pull request ("true" or "false").
         - self.GIT_COMMIT: The commit ID of the GIT CI build.
         """
+        try:
+            os.chdir(self.ValidationPath)
 
-        os.chdir(self.ValidationPath)
-
-        webpage_path = os.path.join(self.ValidationPath, "Webpage")
+            webpage_path = os.path.join(self.ValidationPath, "Webpage")
+            if not os.path.isdir(webpage_path):
+                subprocess.run(["git", "clone", f"https://{self.VALIDATION_GIT_PATH}", "--single-branch", "--depth", "1", "-b", self.WEBPAGE_BRANCH, self.WEBPAGE_FOLDER])

```

Comment:
Not a full review yet (I promise I'll find some time!) but thought I'd draw your attention to this snippet of a WCSim validation failure (with the current code)

```
2024-04-19T14:28:46.9852402Z Cloning into './Webpage'...
2024-04-19T14:31:00.8269582Z fatal: unable to access 'https://github.com/WCSim/Validation.git/': Failed to connect to github.com port 443: Connection timed out
2024-04-19T14:31:00.8297409Z Traceback (most recent call last):
2024-04-19T14:31:00.8298478Z   File "/opt/Validation/RunTests.py", line 22, in <module>
2024-04-19T14:31:00.8300576Z     cw.checkout_validation_webpage_branch()
2024-04-19T14:31:00.8303885Z   File "/opt/Validation/common.py", line 52, in checkout_validation_webpage_branch
2024-04-19T14:31:00.8304968Z     os.chdir("Webpage")
2024-04-19T14:31:00.8305944Z FileNotFoundError: [Errno 2] No such file or directory: 'Webpage'
```

Looks like a bad net connection on one of the runners. I don't know the cause of this, so I don't know if it's a good idea to "wait 2 minutes & try again", but something to think about

I think I have seen this in the past, but it is rare (maybe 1 in 1000 subjobs) so is workable without a fix (github actions have a "re-run failed jobs"). But would be nice to try handle it in the code (I think a similar solution should be tried for all `git push/pull/clone` commands)

Reply:
I agree, this is something that can be easily caught in the code and avoids re-running the full job on failure. Will implement!